assets/pages/weekly-subscription-landing/components/weeklyForm.jsx

```diff
@@ -2,90 +2,69 @@
-        const {
-          copy, title,
-        } = billingPeriods[type];
-        return (
-          <div className="weekly-form__item">
-            <WeeklyFormLabel
-              title={title}
-              offer={billingPeriods[type].offer || null}
-              type={type}
-              key={type}
-              checked={type === selectedPeriod}
-              onChange={() => { setPeriodAction(type); }}
-            >
-              {copy(countryGroupId)}
-            </WeeklyFormLabel>
-          </div>
-          );
-        })}
-      </div>
-    </div>
-    <div className={['weekly-form__cta', bgClassName].join(' ')} data-disabled={selectedPeriod === null}>
-      <WeeklyCta disabled={selectedPeriod === null} type="submit">
-        Subscribe now{selectedPeriod && ` – ${billingPeriods[selectedPeriod].title}`}
-      </WeeklyCta>
-    </div>
-
-    <div className="weekly-form__info">
-      <SvgInfo />
-      You can cancel your subscription at any time
-    </div>
-  </form>
-);
+import { currencies, detect } from 'helpers/internationalisation/currency';
+import { type WeeklyBillingPeriod, getWeeklyProductPrice } from 'helpers/subscriptions';
+import { type Action } from 'components/productPage/productPagePeriodForm/productPagePeriodFormActions';
+import ProductPagePeriodForm, { type StatePropTypes, type DispatchPropTypes } from 'components/productPage/productPagePeriodForm/productPagePeriodForm';
+
+import { type State } from '../weeklySubscriptionLandingReducer';
+import { redirectToWeeklyPage, setPeriod } from '../weeklySubscriptionLandingActions';
+
+
+// ---- Periods ----- //
+
+const getPrice = (countryGroupId: CountryGroupId, period: WeeklyBillingPeriod) => [
+  currencies[detect(countryGroupId)].extendedGlyph,
+  getWeeklyProductPrice(countryGroupId, period),
+].join('');
+
+export const billingPeriods = {
+  sixweek: {
+    title: '6 for 6',
+    offer: 'Introductory offer',
+    copy: (countryGroupId: CountryGroupId) => `${getPrice(countryGroupId, 'sixweek')} for the first 6 issues (then ${getPrice(countryGroupId, 'quarter')} quarterly)`,
+  },
+  quarter: {
+    title: 'Quarterly',
+    copy: (countryGroupId: CountryGroupId) => `${getPrice(countryGroupId, 'quarter')} every 3 months`,
+  },
+  year: {
+    title: 'Annually',
+    copy: (countryGroupId: CountryGroupId) => `${getPrice(countryGroupId, 'year')} every 12 months`,
+  },
+};
+
 
 // ----- State/Props Maps ----- //
 
-const mapStateToProps = (state: State) => ({
-  selectedPeriod: state.page.period,
-  countryGroupId: state.common.internationalisation.countryGroupId,
-});
+const mapStateToProps = (state: State): StatePropTypes<WeeklyBillingPeriod> => {
+  const periods = {};
+  Object.keys(billingPeriods).forEach((k) => {
+    periods[k] = {
+      title: billingPeriods[k].title,
+      copy: billingPeriods[k].copy(state.common.internationalisation.countryGroupId),
+      offer: billingPeriods[k].offer || null,
+    };
+  });
```

Comment:
Optionally, if you wanted to do this without creating an intermediate mutable object I think `reduce` may work:

```jsx
const periods = Object.keys(billingPeriods).reduce((ps, k) => ({
  ...ps,
  [k]: {
    title: billingPeriods[k].title,
    copy: billingPeriods[k].copy(state.common.internationalisation.countryGroupId),
    offer: billingPeriods[k].offer || null,
  },
}), {});
```

Reply:
omg this is great!